### PR TITLE
feat(defect report): new form with image upload

### PR DESCRIPTION
**Pull request by donni106**
- created new feature form to report defects
- refactored `ImageSelector` with a `isConsul` in addition to `isVolunteer` and added a default for our main server
- refactored location settings abort button to be the same as in the new screen being less significant

you need to add a new service icon in homeService for example
```
{
  "title": "Mängelmelder",
  "routeName": "Html",
  "iconName": "warning-outline",
  "params": {
    "title": "Mängelmelder",
    "rootRouteName": "Mängelmelder",
    "query": "publicHtmlFile",
    "queryVariables": {
      "name": "defectreport"
    },
    "subQuery": {
      "routeName": "DefectReportForm",
      "buttonTitle": "Mängel melden",
      "params": {
        "rootRouteName": "Mängelmelder",
        "consentForDataProcessingText": "Maecenas sed diam eget risus varius blandit sit amet non magna. Aenean lacinia bibendum nulla sed consectetur. Nullam quis risus eget urna mollis ornare vel eu leo"
      }
    }
  }
}
```

you also need to create a new category with some sub categories.

image

the main category id needs to be put in global settings:

image

SVA-738

Simulator Screen Shot - iPhone 12 mini - 2022-12-15 at 15 15 29-tile

### CODE REVIEW

**Comment by donni106** [comment]
Category selection added

Simulator Screen Shot - iPhone 12 mini - 2022-12-15 at 15 55 53

File: src/components/defectReport/DefectReportLocationForm.tsx [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    if (
+      shouldGetPosition &&
+      !loadingPosition &&
+      !loadingLastKnownPosition &&
+      (position || lastKnownPosition)
+    ) {
+      setIsLocationSelect(false);
+      setSelectedPosition((position || lastKnownPosition)?.coords);
+    }
+  }, [lastKnownPosition, loadingPosition, loadingLastKnownPosition, position, shouldGetPosition]);
```
Comment (ardasnturk) [inline src/components/defectReport/DefectReportLocationForm.tsx:53]: If the location permissions in the system are turned off, the location value is `undefined`. 

Is it supposed to be like this? Or should we ask the user to grant location permission?

File: src/components/defectReport/DefectReportCreateForm.tsx [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+                  name: 'images',
+                  label: texts.volunteer.images,
+                  buttonTitle: texts.volunteer.addImage
+                }
+              }}
+            />
+          )}
+          control={control}
+        />
+      </Wrapper>
```
Comment (ardasnturk) [inline src/components/defectReport/DefectReportCreateForm.tsx:208]: After selecting an image, I get a screen like this. The delete image button is not visible on the screen and is not centred exactly.

IMG_D5142462639D-1

File: src/components/defectReport/DefectReportCreateForm.tsx [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+              phone: defectReportNewData.phone
+            }
+          ],
+          contentBlocks: [{ body: defectReportNewData.body, title: defectReportNewData.title }],
+          addresses: [
+            {
+              geoLocation: {
+                latitude: selectedPosition?.latitude,
+                longitude: selectedPosition?.longitude
+              }
```
Comment (ardasnturk) [inline src/components/defectReport/DefectReportCreateForm.tsx:113]: If I press the `Auf der Karte auswählen` button and then press the `weiter` button without selecting a new position on the map, the `selectedPosition` is `undefined`.

image

**Review by ardasnturk — COMMENTED** [review]
I must say it works well. I think it will be much better after a little refactor.

File: src/components/defectReport/DefectReportCreateForm.tsx (hunk 2 above)
Comment (donni106) [reply src/components/defectReport/DefectReportCreateForm.tsx:208]: ok nice catch, we need a condition for wider screens

File: src/components/defectReport/DefectReportCreateForm.tsx (hunk 3 above)
Comment (donni106) [reply src/components/defectReport/DefectReportCreateForm.tsx:113]: very good case, i will have a look and try to set a fallback with the pin we see on the map

File: src/components/consul/selectors/ImageSelector.js [hunk 4]
```diff
@@ -205,14 +237,15 @@ ImageSelector.propTypes = {
...
   isVolunteer: PropTypes.bool,
   item: PropTypes.object
 };
 
 const styles = StyleSheet.create({
   image: {
-    height: imageHeight(imageWidth() * 0.6),
-    width: imageWidth() * 0.6
+    height: imageHeight(imageWidth() * 0.8),
+    width: imageWidth() * 0.8
```
Comment (donni106) [inline src/components/consul/selectors/ImageSelector.js:248]: i will revert this to result in 

image

File: src/components/defectReport/DefectReportLocationForm.tsx (hunk 1 above)
Comment (ardasnturk) [reply src/components/defectReport/DefectReportLocationForm.tsx:53]: I think it's better now.

File: src/components/defectReport/DefectReportCreateForm.tsx (hunk 3 above)
Comment (ardasnturk) [reply src/components/defectReport/DefectReportCreateForm.tsx:113]: Very good 👍

File: src/components/consul/selectors/ImageSelector.js (hunk 4 above)
Comment (ardasnturk) [reply src/components/consul/selectors/ImageSelector.js]: At least the delete button can be pressed. 😂👍

**Review by ardasnturk — APPROVED** [review]

**Pull request merged**